### ISSUE TITLE
feature request: remove task id from (x,y,t) triples when return_task_id=False

### ISSUE THREAD

**Issue by murphyk** [ContinualAI/avalanche#1304]
Currently, when `return_task_id=False`, it still returns the task id, but just sets it to 0.
Thus data iterators need to have the form `for (x,y,t) in iter: ...` whereas most existing code
just uses `for (x,y) in iter: ...`. (See example below). This is super annoying.
In the task agnostic setup, we shouldn't have to  worry about task labels.

```
from avalanche.benchmarks.classic import SplitMNIST


n_experiences = 5
split_mnist = SplitMNIST(n_experiences=n_experiences, seed=1, return_task_id=False,
                         class_ids_from_zero_in_each_exp=True, fixed_class_order=range(10))

train_set = split_mnist.train_stream[0].dataset
train_dataloader = DataLoader(train_set, batch_size=10, shuffle=False)
train_iter = iter(train_dataloader)

(Xb, Yb, Tb) = next(train_iter)
print(Xb.shape) # (10,1,28,28)
print(Tb) # all 0s - no useful information

```

**Comment by AntonioCarta** [comment]
Thanks for the request. We already made some changes so that benchmarks, datasets, and training strategies don't need triplets for task-agnostic benchmarks. 

The problem is that the high-level benchmark generators are still adding the task label by default. We are going to change their behavior in the future.

**Comment by AntonioCarta** [comment]
This is fixed now. We added new benchmark generators that don't add task labels, such as https://github.com/ContinualAI/avalanche/blob/67d8766f3133eb4bce986423355268e69f339e14/avalanche/benchmarks/scenarios/supervised.py#L37
